### DISCUSSION TITLE
Error when evaluating 'environment' for job 'docker'. 'StringToken' was excepted

### DISCUSSION THREAD

**Discussion by tristanbes** [community/community#26725] [Actions]
<p>Hello,</p>
<p>2 hours ago, every job running in the CI worked fine.<br>
Now without touching anything CI related (untouched environements, or CI files), Github actions fails with this error:</p>
<p><strong>Error when evaluating ‘environment’ for job ‘docker’. Unexpected type ‘BasicExpressionToken’ encountered while reading ‘job environment name’. The type ‘StringToken’ was expected.</strong></p>
<p>Part of our file is:</p>
<pre><code class="lang-auto">name: Deploy app

on:
    pull_request:
        types: [opened, synchronize, reopened, ready_for_review]
    push:
        tags:
          - "staging"
        branches:
          - "main"

#concurrency:
#    group: ${{ github.head_ref }}
#    cancel-in-progress: true

env:
    CYPRESS_INSTALL_BINARY: 0

jobs:
    environment:
        name: Set environment
        runs-on: ubuntu-latest
        outputs:
          github_env: ${{ steps.github_env.outputs.env }}
          app_env: ${{ steps.app_env.outputs.env }}
          app_debug: ${{ steps.app_debug.outputs.app_debug }}
          tag: ${{ steps.tag.outputs.tag }}
          host_header: ${{ steps.host_header.outputs.host_header }}
          url: ${{ steps.url.outputs.url }}
          tf_workspace: ${{ steps.tf_workspace.outputs.tf_workspace }}
          tf_pr_number: ${{ steps.tf_pr_number.outputs.tf_pr_number }}
        steps:
          - id: github_env
      [redacted....]
    docker:
        name: Docker build
        needs: [environment]
        runs-on: ubuntu-latest
        permissions:
            contents: read
            packages: write
        environment: 
            name: ${{needs.environment.outputs.github_env}}
        steps:
            - name: Checkout code
              uses: actions/checkout@v2
            - name: Check if image exist
              id: check
    [redacted...]
</code></pre>
<p>the rest has been redacted</p>
<p>Has Github broken things ?</p>

**Comment by rbayliss** [comment]
<p>We’re also experiencing this issue as of an hour or so ago.  This has broken deployments for our project.</p>

**Comment by ethomson** [comment, marked as answer]
<p>Apologies that you’re running into this. This is a result of a bug in a deployment that a small number of workflows will encounter. We’re rolling this back.</p>

**Comment by ethomson** [comment]
<p>This should be fixed. Please let us know if you’re still seeing any problems.</p>

**Comment by ahrelj01** [comment]
<p>We’re seeing the same issue today, since the past 2 hours. Interestingly enough, creating a new repository with the same workflow works without an issue:</p>
<pre><code class="lang-auto">name: PR Open/Sync Validation

on:
  pull_request:
    branches: [development, production]
    types: [opened, synchronize]

jobs:
  feature-branch-change:
    name: feature-branch-change
    runs-on: ubuntu-latest
    if: ${{ github.event.action == 'synchronize' &amp;&amp; github.event.pull_request.base.ref != 'development' }}
    environment: ${{ github.event.pull_request.base.ref }}
    steps:
      - name: Checkout Feature
        uses: actions/checkout@v2
        with:
          token: ${{ secrets.SEMANTIC_RELEASE_ACTION_TOKEN }}
          ref: ${{ github.event.pull_request.head.ref }}
          fetch-depth: 0
</code></pre>
<p>When using</p>
<pre><code class="lang-auto">    environment: ${{ github.event.pull_request.base.ref }}
</code></pre>
<p>the error says <code>Error when evaluating 'environment' for job 'feature-branch-change'. Unexpected type 'BasicExpressionToken' encountered while reading 'job environment name'. The type 'MappingToken' was expected.</code></p>
<p>When using</p>
<pre><code class="lang-auto">    environment: 
      name: ${{ github.event.pull_request.base.ref }}
</code></pre>
<p>the error says <code>Error when evaluating 'environment' for job 'feature-branch-change'. Unexpected type 'BasicExpressionToken' encountered while reading 'job environment'. The type 'MappingToken' was expected.</code></p>
<p>Hope this helps</p>
